docs/source/reference/ha.rst

```diff
@@ -0,0 +1,301 @@
+
+n st2resultstracker will co-operate with each other to perform work. At startup though there is a possibility
+of work however there will not be any negative consequences to this duplicate work. Specifically the jobs to track results
+also get stored in the DB in case there is no worker to take over the work and this pattern makes all result trackers
+pick up the same initial work set. Once this work set is exhausted all subsequents tasks are round-robined.
+
+st2notifier
+^^^^^^^^^^^
+This is a dual purpose process - its main function is to generate ``st2.core.actiontrigger`` and ``st2.core.notifytrigger``
+based on the completion of ActionExecution. The auxiliary purpose is to act as a backup scheduler for actions that may
+not have been scheduled.
+
+n st2notifiers can run in active-active requiring connections to RabbitMQ and MongoDB. For the auxiliary purpose to function
+in an HA deployment when more than 1 st2notifiers are running either Zookeeper or Redis is required to provide co-ordination
+much like for policies. It is also possible to designate a single st2notifier as provider of auxiliary functions by
+disabling the scheduler in all but 1 st2notifiers.
+
+st2garbagecollector
+^^^^^^^^^^^^^^^^^^^
+Cleans up old executions and other operations data based on setup configurations. By default this process does nothing
+and needs to be setup to perform any work.
+
+By design it is a singleton process. Running multiple in active-active will not yield much benefit and also will not
+do any harm. Ideal configuration is active-passive but |st2| does not itself provide ability to run this in active-passive.
+
+mistral-api
+^^^^^^^^^^^
+Mistral api is served by this aptly named process. It needs access to PostgreSQL and RabbitMQ.
+
+n mistral-api can run and just like st2api in active-active configuration by using a loadbalancer to distribute at its
+front end. In typical single box deployment mistral-api is local to the box and |st2| communicates via a direct HTTP
+connection however for HA setup we recommend putting mistral-api behind a loadbalancer and setting up |st2| to communicate
+via the loadbalancer.
+
+mistral-server
+^^^^^^^^^^^^^^
+mistral-server is the worker engine for mistral i.e. the process which actually manages executions. |st2| plugin to
+mistral i.e. ``st2mistral`` communicates back to |st2| api. This process needs access to PostgreSQL and RabbitMQ.
+
+n mistral-server can run and co-ordinate work in an active-active configuration. In an HA deployment all communication
+with the |st2| API must be via the configured loadbalancer.
+
+Required dependencies
+---------------------
+Some HA recommendations for the dependencies required by |st2| components. Depending on the exact infrastructure these
+may not be suitable and would only serve as a suggestion.
+
+MongoDB
+^^^^^^^
+|st2| uses this to cache Actions, Rules and Sensor metadata which already live in the filesystem. All the content should
```

Comment:
I would probably say something along the lines of that is mostly servers the purpose of storing meta data.